Create vtex-shipping-network-labels-integration.md (#275)
vtexdocs/dev-portal-content: docs/guides/Integration-Guides/vtex-shipping-network/vtex-shipping-network-labels-integration.md

---
title: "VTEX Shipping Network Labels Integration"
slug: "vtex-shipping-network-labels-integration"
hidden: true
createdAt: "2023-03-01T16:44:01.460Z"
updatedAt: "2023-03-01T16:44:01.460Z"
---

[VTEX Shipping Network](https://vtex.com/br-pt/shipping-network/) is the solution that uses order tracking data directly from [carriers](https://help.vtex.com/en/tutorial/carries-on-vtex--7u9duMD5UQa2QQwukAWMcE) to keep merchants and shoppers up to date with the status of each delivery.

## Overview

The VTEX Shipping Network module is responsible for integrating your store with carriers via a defined protocol. The integration is divided into three workflows:

1. Notification
2. Tracking
3. Shipping label creation

> The following lifecycle overview of a Package created in the VTEX Shipping Network module (SPN) will provide an overview of these workflows:

Reviewer: ```suggestion
 A Package created in the VTEX Shipping Network module (SPN) has the following lifecycle, including these workflows:
```

1. The cycle begins when an order is invoiced in the OMS module, which results in SPN creating all of its packages. The new packages will have an `Invoiced` [status](https://help.vtex.com/en/tutorial/order-flow-and-status--tutorials_196).

> 2. The Carrier App will then receive a notification request for all packages, and their status will move to `carrier_notified`.

Reviewer: Dá pra adicionar o link do Carrier App aqui. Tá publicada a doc?

3. All packages can now have their associated Shipping Label created. Once each label is successfully created, its associated package status moves to `label_issued`.
4. Tagged by its Shipping Label, the package can now be delivered, and while it is in transit, the SPN module requests the Carrier App for tracking updates until the package is tagged as `Delivered`.

The Package lifecycle ends when it is delivered.

## Integration

The endpoint described in this document is responsible for creating the Shipping Labels of all the packages associated with an order. Given an `orderId`, it will provide an URL that when accessed downloads a pdf file containing the Shipping Labels.

>❗ The URL returned by the API is valid for 5 days from the moment it was requested, and expires after that period. If the file is required after the link has expired, only a new request with the same `orderId` will provide a new valid URL.

> Be aware that VTEX Shipping Network only keeps track of Fulfillment orders, therefore, the `orderId` used on the endpoint must have their origin as Fulfillment. All Fulfillment Orders contain a prefix, for example:

Reviewer: Temos link pra explicar mais sobre o que são fulfillment orders?

Reviewer: Tinha pensado em algo como [aqui](https://developers.vtex.com/docs/guides/multilevel-omnichannel-inventory#order-flow), mas não tem a terminologia de Fulfillment orders nessa doc de MOI

Author: Feito.
